tests/framework/db/ColumnSchemaTest.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace yiiunit\framework\db;
+
+use yiiunit\framework\db\DatabaseTestCase;
+use yii\db\ColumnSchema;
+
+/**
+ * @group db
+ * @group mysql
+ */
+class ColumnSchemaTest extends DatabaseTestCase
+{
+
+    public function testDbTypecastNoCommas()
+    {
+        $locale = setlocale(LC_NUMERIC, 0);
+        if (false === $locale) {
+            $this->markTestSkipped('Your platform does not support locales.');
+        }
+
+        try {
+            // This one sets decimal mark to comma sign
+            setlocale(LC_NUMERIC, 'ru_RU.utf8');
+            $cschemaDouble = new ColumnSchema([
+                'type' => 'double',
+                'dbType' => 'double',
+                'phpType' => 'double',
+            ]);
+            $this->assertEquals('1.23', (string) $cschemaDouble->dbTypecast('1.23'));
+            $this->assertEquals('1.23', (string) $cschemaDouble->dbTypecast(1.23));
+            $this->assertEquals('2.0E+30', (string) $cschemaDouble->dbTypecast(2e+30));
```

Comment:
Yeah, there is some casting to string inside of PDO, thats why I wrote the test in #14448 with writing to the DB, turns out the problem is not in ColumnSchema, but in QueryBuilder::batchInsert(). Thanks for the feedback!